Code/GraphMol/Descriptors/PMI.cpp

```diff
@@ -0,0 +1,111 @@
+//
+//  Copyright (C) 2016 Greg Landrum
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MolTransforms/MolTransforms.h>
+#include "PMI.h"
+#include <Eigen/Dense>
+
+namespace RDKit {
+namespace Descriptors{
+namespace {
+
+bool getMoments(const ROMol& mol,int confId, bool useAtomicMasses,
+                double &pm1, double &pm2, double &pm3){
+  PRECONDITION(mol.getNumConformers()>=1,"molecule has no conformers");
+  const char *pn1 = useAtomicMasses ? "PMI1_mass" : "PMI1";
```

Comment:
I have been adding them to a local namespace to prevent typos in these cases.  It also makes it easier to move into the global later.